src/Illuminate/Routing/Route.php

```diff
@@ -626,7 +626,11 @@ public function setUri($uri)
      */
     public function getName()
     {
-        return $this->action['as'] ?? null;
+        return isset($this->action['as'])
+            ? Str::endsWith($this->action['as'], '.')
+                ? substr($this->action['as'], 0, -1)
+                : $this->action['as']
```

Comment:
You could `rtrim($this->action['as'], '.')`, so not need for the extra condition. :)